junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorService.java

```diff
@@ -175,4 +176,22 @@ public void compute() {
 
 	}
 
+	static class WorkerThreadFactory implements ForkJoinPool.ForkJoinWorkerThreadFactory {
+
+		private final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+
+		@Override
+		public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+			return new WorkerThread(pool, contextClassLoader);
+		}
+	}
+
+	static class WorkerThread extends ForkJoinWorkerThread {
+
+		WorkerThread(ForkJoinPool pool, ClassLoader contextClassLoader) {
+			super(pool);
+			super.setContextClassLoader(contextClassLoader);
```

Comment:
nit: `super` is unnecessary here, isn't it?

Reply:
Followed the example of the `DefaultWorkerThread` from the JDK...

Reply:
```suggestion
			setContextClassLoader(contextClassLoader);
```